docs/FRSKY_R9.md

```diff
@@ -64,37 +84,105 @@ The wiring is similar to the ExpressLRS docs found in the link below. Different
 
-In these docs it is suggested to download and use the "ST-LINK Utility" software. This software is pretty outdated (NRND = not recommended for new designs), and the new recommended tool is "STM32CubeProgrammer". However, there might be catches:
-- ST-LINK Utility appears to not work with newer ST-Link programmers.
-- STM32CubeProgrammer/STM32CubeIDE is quite nasty with which ST-Link programmer is used. It seems ST really wants their tools to only work well with "original" or legit ST-Link programmers. I especially had significant issues with getting the cheap and widely available 8$ STLinkV2 usb-stick-like clones to work with STM32CubeProgrammer/STM32CubeIDE, and they then never worked reliably or to my satisfaction.
-- It seems it can happen that ST-LINK Utility and STM32CubeProgrammer/STM32CubeIDE do not like each other.
-- STM32CubeProgrammer runs per default at a high SWD frequency, like 4000 kHz, on the 8$ STLinkV2 clones. Setting the frequency manually to 480 kHz and using short wires to the device makes it work more reliable.
-- STM32CubeProgrammer will likely require new firmware be installed on the ST-Link V2 or compatible device via the "Firmware upgrade" button" the first time it is used.
+4. Scroll to the FIRMWARE folder and select the r9m\_elrs\_bl.frk file.  Choose "Flash external module".
 
-So, if you use or want to use one of these 8$ STLinkV2 usb-stick-like clones and you have trouble with STM32Cube, you probably want to install ST-LINK Utility and see how that works for you. You will then (likely) not be able to flash from STM32CubeIDE, but have to go with the ST-LINK Utility.
+5. Follow the instructions in the next section to flash mLRS.
 
-If anyone has deeper/better insight into the STM32CubeProgrammer/STM32CubeIDE vs ST-LINK Utility and cheap STLinkV2 programmer clones issues, please help :)
+If you have already flashed via ST-Link, you can continue to flash updates via ST-Link the way you always have, or you can follow these steps to switch to flashing from the radio.  To do this, you will flash the ELRS bootloader by using ST-Link one last time:
 
-If you wonder what STLink programmer I (olliw42) am using: Every NUCLEO board comes also with a STLink programmer... and these board are relatively cheap too, about 15Eur.
+1. Download the [r9m\_bootloader.bin](https://github.com/ExpressLRS/ExpressLRS/blob/master/src/bootloader/r9m_bootloader.bin?raw=true) file from the ExpressLRS git repository.
+2. Flash it to the beginning of the flash (0x8000000) using ST-Link and STM32CubeProgrammer.
+
+3. Follow the instructions in the next section to flash mLRS.
+
+#### Flash/Update the mLRS firmware ####
+
+Once the ELRS bootloader is installed, the following precedure can be followed to install the mLRS firmware.
+
+1. Download the latest mLRS firmware or build with MLRS\_FEATURE\_ELRS\_BOOTLOADER defined.
+
+2. Copy the latest tx-R9M-f103c8-elrs-bl-r9m-v*.elrs file to FIRMWARE folder on your radio's SD card.
```

Comment:
- the name is still incorrect, isn't it? (no -r9m)
- doesn't it make more sense to tell the name of the file in point 1.? Like "1. Download the latest mLRS firmware file tx-R9M-f103c8-elrs-bl-vX.XX.XX.elrs or build..." 
- I guess I would add a "... (note the 'elrs-bl' in the file name)"

Reply:
I think they might want to download the entire repository so it makes more sense to keep the first mention of the file name where it is.

Reply:
file name still needs correction, IMHO :)